controllers/metal3.io/baremetalhost_controller.go

```diff
@@ -717,6 +721,11 @@ func (r *BareMetalHostReconciler) actionInspecting(prov provisioner.Provisioner,
 		return result
 	}
 
+	// Indicate that the hostFirmwareSettings resource should be updated by its controller
+	if err = r.markHostFirmwareSettingsForUpdate(info); err != nil {
```

Comment:
On a second thought I'm afraid that this approach won't work properly, as it doesn't seem enough to cope with the BareMetalHost lifecycle: as per the last discussions, every BMH should have its own HFS (and schema) - and moreover I guess that it should be deleted as long as the related BMH disappear.
That's also something to consider during an upgrade, where some pre-existing BMHs could exist withouth ofc any related HFS (and then should be created).
Given that, I think that a viable approach  could be to let the HFS controller watching also the BMH events, and move directly the logic for the creation (and update) directly in the HFS controller, trying to let the BMH controller just a simple reader of the HFS resources (just for building the cleaning steps).
In this way the relationship will be come a little bit more loosely coupled (even though I guess a sync point will still be required, at least to let understand the BMH that the its own HFS has been filled properly).

Reply:
That's a really good point. I've removed the creation of the hfs and the setting of the update flag from the bmh and added event handlers to the hfs controller and Watches() for the bmh. This is cleaner and has reduced the coupling between the 2 controllers. The bmh just reads the associated hfs resource to get the clean steps.